### DISCUSSION TITLE
Fantastische Datenbanken und wo sie zu finden sind

### DISCUSSION THREAD

**Discussion by FlorianHeck2** [FlorianHeck2/codespaces-jupyter#7] [Announcements]
Wie der Titel schon sagt, so oder so ähnlich denken wir wohl alle über Datenbanken - gibt es was schöneres? Die einzig wahre Antwort ist ein klares NEIN!
Damit ihr dazu auch einen Bezug habt, warum sie so praktisch sind, fass ich es euch zusammen, dann müsst ihr euch nicht damit beschäftigen :)
In Python ist es sehr einfach, Datenbanken anzulegen. Dafür muss man sqlite3 importieren und über sqlite3.connect("NameDerDatnbank.db") mit einer Datenbank verbinden. und einen Cursor erstellen, über den man die SQL-Befehle eingeben kann: cur = con.cursor()
Über diesen Cursor kann man nun SQL-Befehle eingeben und so einfach eine Datenbank aufbauen, in dieser Daten speichern und wieder abrufen, was sich vor allem für größere Tabellen lohnt. 
Für einen Einblick zum Ausprobieren kann ich folgendes [Tutorial](https://docs.python.org/3/library/sqlite3.html) empfehlen.
Ich habe dabei in zwei Dateien, einer zum [Füllen](https://github.com/FlorianHeck2/codespaces-jupyter/blob/main/sqltests.py) und einer zum [Lesen](https://github.com/FlorianHeck2/codespaces-jupyter/blob/main/sqlreader.py) der Datenbank gearbeitet.

**Comment by Rouge2123** [comment]
Das finde ich echt cool! Vielen Dank, dass du auch eine Anleitung dazu geschrieben hast. Es hört sich fast zu einfach an 👀

**Reply by FlorianHeck2** [reply]
Gerne; ja, es ist echt einfach (gerade, wenn man bereits etwas sql beherrscht), kann ich nur empfehlen.

**Comment by 1Wargear** [comment]
Interessant das das so einfach mit den Datenbanken geht, gibt es auch einen Object-Relation-Mapper (ORM)?

**Reply by FlorianHeck2** [reply]
Damit hab ich mich noch nicht beschäftigt, ich wollte aber mal anfangen, eine Datenbank zu nutzen, um Speicherstände eines Spiels zu erstellen. Da würde das dann mit rein spielen. Wenn ich was dazu finde poste ich das vsl nächste Woche :)
